src/lockscreen.js

```diff
@@ -130,6 +142,27 @@ var Lockscreen = GObject.registerClass({
         this._updateBackground();
     }
 
+    _playActiveSound() {
+        if (!this._locked)
+            return;
+        if (!this._playbin.hasKey)
+            return;
+        if (this._activeSoundID === 'pending')
+            return;
+        this._activeSoundID = 'pending';
```

Comment:
I think for this to work correctly you have to do the other half of it as well, like we do in the shell; check for `pending` in `_stopActiveSound` and set it to `cancelled`, and check for `cancelled` here as well.

Reply:
I am actually trying something completely different in parallel now, since this UX has one problem I discussed here https://phabricator.endlessm.com/T24801#658274, and yeah, what I am trying now kind of uses that, but I am moving it to the sound wrapper.